### PR TITLE
added zoid component behaviour

### PR DESCRIPTION
**Pull request by poyamz**
The logic which manages the zoid intergration

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+require('dotenv').config();
+const path = require('path');
+const webpack = require('webpack');
+
+const setupDomainUrl = () => {
+    let domainUrl;
+    switch (process.env.STAGING) {
+        case '0':
```
Comment (ardation) [inline webpack.config.js:9]: I think this needs to take into consideration now PR builds. You can get that from NOW_URL Environment variables.

File: pages/widget.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        };
+    }
+}
+type Xprops = {
+    xprops: {
+        countryCode: string;
+    };
+};
+
+class WidgetPage extends Component<GetCountriesAndSubdivisions, Xprops> {
```
Comment (ardation) [inline pages/widget.tsx:23]: possible to convert this to a function component?

File: package.json [hunk 3]
```diff
@@ -13,7 +13,10 @@
         "test": "jest",
         "test:watch": "jest --watch",
         "test:coverage": "jest --coverage",
-        "chromatic": "chromatic"
+        "chromatic": "chromatic",
+        "build-widget-dev": "STAGING=-1 npx webpack --config webpack.config.js",
+        "build-widget-stage": "STAGING=1 npx webpack --config webpack.config.js",
+        "build-widget-prod": "STAGING=0 npx webpack --config webpack.config.js"
```
Comment (ardation) [inline package.json:19]: is it possible to rely on environment variables in NOW to configure which build should occur?

Comment (poyamz) [reply package.json:19]: That's what I'm doing for 'stage' and 'prod' - I've assigned the STAGING env in the NOW config so in theory they can be removed from package. I'd need to keep something to demarcate the env for the dev build though.

File: pages/widget.tsx (hunk 2 above)
Comment (poyamz) [reply pages/widget.tsx:23]: In theory yes. At first I tried to keep it as a functional component but I was struggling to access or pass in the window.xprops property. I was actually wondering if you'd have a suggestion as to how best to achieve that?

File: webpack.config.js (hunk 1 above)
Comment (poyamz) [reply webpack.config.js:9]: Ah okay, I'll have a look into that.

Comment (poyamz) [reply webpack.config.js:9]: I've added the NOW_URL system environement env to the now build config and have updated the webpack.config.js to use them if available and to default to the current urls if not.

File: package.json (hunk 3 above)
Comment (poyamz) [reply package.json]: I've removed all 3 of the build-widget-* tasks and have placed the npx call for build-widget-dev into the npm run dev command directly. I have also updated the Now environment env for developement

**Pull request merged**